packages/tupaia-web/src/layout/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useContext, useEffect } from 'react';
+import styled from 'styled-components';
+import {
+  MAX_SIDEBAR_EXPANDED_WIDTH,
+  TRANSPARENT_BLACK,
+  CONTROL_BAR_PADDING,
+  CONTROL_BAR_WIDTH,
+  MAP_CONTROLS_WIDTH,
+} from '../../theme';
+import { ExpandButton } from './ExpandButton';
+import { SIDEBAR_ACTION_TYPES, SidebarContext, SidebarDispatchContext } from '../../context';
+
+const MIN_DEFAULT_WIDTH = 350;
+const MAX_DEFAULT_WIDTH = 500;
+const DEFAULT_WIDTH_RATIO = 0.2;
+
+const Panel = styled.div<{
+  $width: number;
+}>`
+  display: flex;
+  align-items: stretch;
+  flex-direction: column;
+  align-content: stretch;
+  max-width: ${MAX_SIDEBAR_EXPANDED_WIDTH}px;
+  position: relative;
+  overflow: visible;
+  background-color: ${TRANSPARENT_BLACK};
+  pointer-events: auto;
+  height: 100%;
+  cursor: auto;
+  transition: width 0.5s ease;
+  width: ${({ $width }) => $width}px;
+`;
+
+export const Sidebar = () => {
+  const { isExpanded, expandedWidth, contractedWidth } = useContext(SidebarContext);
+  const dispatch = useContext(SidebarDispatchContext);
+
+  // Update the sidebar widths when the window is resized
+  const updateDimensions = () => {
```

Comment:
I think this logic from the old project is a bit too calculated. I think that the sidebar width should just be done using responsive css and then the expanded width can just be a bigger ratio or percentage width. There will need to be some responsive styles set at somepoint too but that can be done later.